src/utils/system_utils/tests/system_utils_ut.cpp

```diff
@@ -330,9 +329,13 @@ TEST_CASE_METHOD(TestCaseFixture, "SystemUtils_ForEachDir")
 
         const std::string test_subdir = "foo_subdir";
 
-        std::stringstream ss;
-        ss << "42, " << TestPath() << ", " << test_subdir << std::endl;
-        std::string expected = ss.str();
+        std::vector<std::string> expected;
+
+        {
+            std::stringstream ss;
+            ss << "42, " << TestPath() << ", " << test_subdir << std::endl;
+            expected.push_back(ss.str());
+        }
```

Comment:
im confused. you have a vector of expected paths, but you're still using a comma separated stringsteram here?

Reply:
This internal, test string format in perDirFn callback, is just capturing the 3 arguments being passed to it by SystemUtils_ForEach.

Each time the perDirFn is called, it will push a vector of the 3 args concatenated in this format.

Reply:
I've created a ForEachDirCallRecord struct to explicitly model the callback call capture instead of string concatenation.